analog_forecast.py

```diff
@@ -0,0 +1,297 @@
+"""Script to look for analogs in ERA5 datasets and return RMSE and dates. 
+"""
+
+import argparse
+import xarray as xr
+from dask.distributed import Client
+import pandas as pd
+import numpy as np
+#local
+import luts
+from config import data_dir
+
+
+def parse_args():
+    """Parse some arguments"""
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "-v",
+        dest="varname",
+        type=str,
+        help=(
+            "Variable name to look for analogs. "
+            "Options are t2m (temperature), sst (sea surface temperature), msl (mean sea level pressure)"
+        ),
+        required=True
+    )
+    parser.add_argument(
+        "-d", dest="ref_date", type=str, help="Date to use for analog search", required=True
```

Comment:
Consider including a help line for date arg, to specify date string format. I see that this ```ref_date``` variable is parsed via ```pd.datetime``` at line 113, but I wonder if there are any user provided date strings that could break this....like Euro formatting for single digit day/month (eg 6/1/2023 vs 1/6/2023). 

The ```ref_date``` variable is not parsed in the ```run_rmse_over_time``` function at line 171/172, but you actually do include a help line with desired date format in the ```find_analogs``` function at line 213 and in the ```make_forecast``` function at line 251.